pytest is currently failing

Fixing this is also a precursor to implementing #23.

Currently pytest fails for me. Here is the log:
```
============================= test session starts ==============================
platform linux -- Python 3.10.5, pytest-7.1.2, pluggy-1.0.0 -- /home/alex/code/virtual_rainforest/.venv/bin/python
cachedir: .pytest_cache
rootdir: /home/alex/code/virtual_rainforest, configfile: setup.cfg
plugins: mypy-0.9.1, flake8-1.1.1, mock-3.8.2, cov-3.0.0
collecting ... collected 43 items

docs/source/conf.py::flake-8::FLAKE8 SKIPPED (file(s) previously pas...) [  2%]
docs/source/conf.py::mypy PASSED                                         [  4%]
docs/source/conf.py::mypy-status PASSED                                  [  6%]
docs/source/development/training/mfm.py::flake-8::FLAKE8 SKIPPED (fi...) [  9%]
docs/source/development/training/mfm.py::mypy PASSED                     [ 11%]
docs/source/development/training/mfm.py::mfm.TimesTable PASSED           [ 13%]
docs/source/development/training/mfm.py::mfm.TimesTable.table PASSED     [ 16%]
docs/source/development/training/mfm.py::mfm.my_float_multiplier FAILED  [ 18%]
docs/source/development/training/mfm.py::mfm.my_float_multiplier2 PASSED [ 20%]
docs/source/development/training/mfm.py::mfm.my_float_multiplier3 PASSED [ 23%]
docs/source/development/training/mfm.py::mfm.my_picky_float_multiplier PASSED [ 25%]
docs/source/development/training/mfm.py::mfm.my_validated_float_multiplier FAILED [ 27%]
docs/source/development/training/test_mfm.py::flake-8::FLAKE8 SKIPPED    [ 30%]
docs/source/development/training/test_mfm.py::mypy PASSED                [ 32%]
docs/source/development/training/test_mfm.py::test_fm PASSED             [ 34%]
docs/source/development/training/test_mfm.py::test_pfm PASSED            [ 37%]
docs/source/development/training/test_mfm.py::test_pfm_fail FAILED       [ 39%]
docs/source/development/training/test_mfm.py::test_pfm_param_noid[1.0-3.0-3.0] PASSED [ 41%]
docs/source/development/training/test_mfm.py::test_pfm_param_noid[1.5--3.0--4.5] PASSED [ 44%]
docs/source/development/training/test_mfm.py::test_pfm_param_noid[-1.5-3.0--4.5] PASSED [ 46%]
docs/source/development/training/test_mfm.py::test_pfm_param_noid[-1.5--3.0-4.5] PASSED [ 48%]
docs/source/development/training/test_mfm.py::test_pfm_param_ids[++] PASSED [ 51%]
docs/source/development/training/test_mfm.py::test_pfm_param_ids[-+] PASSED [ 53%]
docs/source/development/training/test_mfm.py::test_pfm_param_ids[+-] PASSED [ 55%]
docs/source/development/training/test_mfm.py::test_pfm_param_ids[--] PASSED [ 58%]
docs/source/development/training/test_mfm.py::test_pfm_twoparam[+-+] PASSED [ 60%]
docs/source/development/training/test_mfm.py::test_pfm_twoparam[+--] PASSED [ 62%]
docs/source/development/training/test_mfm.py::test_pfm_twoparam[--+] PASSED [ 65%]
docs/source/development/training/test_mfm.py::test_pfm_twoparam[---] PASSED [ 67%]
docs/source/development/training/test_mfm.py::test_pfm_twoparam_fixture[+-+] PASSED [ 69%]
docs/source/development/training/test_mfm.py::test_pfm_twoparam_fixture[+--] PASSED [ 72%]
docs/source/development/training/test_mfm.py::test_pfm_twoparam_fixture[--+] PASSED [ 74%]
docs/source/development/training/test_mfm.py::test_pfm_twoparam_fixture[---] PASSED [ 76%]
docs/source/development/training/test_mfm.py::test_times_table_errors PASSED [ 79%]
docs/source/development/training/test_mfm.py::test_times_table_values PASSED [ 81%]
virtual_rainforest/__init__.py::flake-8::FLAKE8 SKIPPED (file(s) pre...) [ 83%]
virtual_rainforest/__init__.py::mypy PASSED                              [ 86%]
virtual_rainforest/version.py::flake-8::FLAKE8 SKIPPED (file(s) prev...) [ 88%]
virtual_rainforest/version.py::mypy PASSED                               [ 90%]
virtual_rainforest/core/__init__.py::flake-8::FLAKE8 SKIPPED (file(s...) [ 93%]
virtual_rainforest/core/__init__.py::mypy PASSED                         [ 95%]
virtual_rainforest/core/grid.py::flake-8::FLAKE8 SKIPPED (file(s) pr...) [ 97%]
virtual_rainforest/core/grid.py::mypy PASSED                             [100%]WARNING: Failed to generate report: No data to report.



=================================== FAILURES ===================================
______________________ [doctest] mfm.my_float_multiplier _______________________
014 Multiplies two floats together.
015 
016     Arguments:
017         x: The first number
018         y: The second number
019 
020     Examples:
021         >>> my_float_multiplier(2.1, 3.6)
Expected:
    7.56
Got:
    7.5600000000000005

/home/alex/code/virtual_rainforest/docs/source/development/training/mfm.py:21: DocTestFailure
_________________ [doctest] mfm.my_validated_float_multiplier __________________
126 Multiplies two floats together.
127 
128     Arguments:
129         x: The first number
130         y: The second number
131 
132     Examples:
133         >>> my_float_multiplier(2.1, 3.6)
Expected:
    7.56
Got:
    7.5600000000000005

/home/alex/code/virtual_rainforest/docs/source/development/training/mfm.py:133: DocTestFailure
________________________________ test_pfm_fail _________________________________

    def test_pfm_fail():
    
>       with pytest.raises(ValueError) as err_hndlr:
E       Failed: DID NOT RAISE <class 'ValueError'>

docs/source/development/training/test_mfm.py:23: Failed

---------- coverage: platform linux, python 3.10.5-final-0 -----------

===================================== mypy =====================================
Success: no issues found in 7 source files
=========================== short test summary info ============================
FAILED docs/source/development/training/mfm.py::mfm.my_float_multiplier
FAILED docs/source/development/training/mfm.py::mfm.my_validated_float_multiplier
FAILED docs/source/development/training/test_mfm.py::test_pfm_fail - Failed: ...
=================== 3 failed, 33 passed, 7 skipped in 0.66s ====================
```

Note that none of the failures are to do with virtual rainforest code! However, you do want ``pytest`` to pass cleanly so that you can automatically check it with your CI system.

There are a few ways you can go about solving this:

1. Exclude the training materials from testing by changing your ``pytest`` config
2. Fix the failures in the training materials
3. Implement #33

--- a/setup.cfg
+++ b/setup.cfg
@@ -6,6 +6,7 @@ addopts =
 	-v --flake8 --mypy -p no:warnings
 	--cov=myproject  --cov-report=html:htmlcov/coverage
 	--doctest-modules --ignore=myproject/__main__.py
+testpaths = test
 
 [pycodestyle]
 max-line-length = 88

